### PR TITLE
Button group enhancements

### PR DESCRIPTION
**Pull request by arinaturcu**
Enhancements on the Button group block

### CODE REVIEW

**Comment by arinaturcu** [comment]
@HardeepAsrani @Soare-Robert-Daniel Do you have any ideas on how to shorten the CSS that I added in this PR?
I'm not really happy with it since there is some duplicated code, so if you think there are thing I can do to write it better let me know.

**Comment by arinaturcu** [comment]
@HardeepAsrani Can you review the code here before handling this to QA?

File: src/blocks/blocks/button-group/group/inspector.js [hunk 1]
```diff
@@ -105,6 +158,28 @@ const Inspector = ({
...
+									key={ option.value }
+									icon={ option.icon }
+									title={ option.label }
+									isPrimary={ option.value === ( attributes.align[ currentDevice ] ?? 'none' ) }
+									showTooltip
+									onClick={ () => onAlignmentChange( option.value ) }
+								/>
+							);
+						}) }
+					</Toolbar>
```
Comment (HardeepAsrani) [inline src/blocks/blocks/button-group/group/inspector.js:181]: You don't need to make a new component here, there's one in Core that you can use already:

```
import { BlockAlignmentToolbar } from '@wordpress/block-editor';

<BlockAlignmentToolbar
	value={ value }
	isCollapsed={ false }
	controls={ [ 'left', 'center', 'right', 'full' ] }
	onChange={ () => {} }
/>
```

**Review by HardeepAsrani — CHANGES_REQUESTED** [review]
I've left a couple of comments on lines.

**Comment by arinaturcu** [comment]
@HardeepAsrani Thank you! Working on it.

File: src/blocks/blocks/button-group/button/inspector.js [hunk 2]
```diff
@@ -86,8 +92,45 @@ const Inspector = ({
...
+		secondary: {
+			label: __( 'Secondary', 'otter-blocks' )
+		}
+	};
+
 	return (
 		<InspectorControls>
+			{ window.themeisleGutenberg.hasNeve &&
+				<PanelBody
+					title={ __( 'Types', 'otter-blocks' ) }
```
[the remark this answers is not in the document]
Comment (arinaturcu) [reply src/blocks/blocks/button-group/button/inspector.js:111]: So there should be 4 styles: fill, outline, primary, and secondary?

**Comment by HardeepAsrani** [comment]
@arinaturcu Yea, if the Neve is activated then we will have four, two of them will just inherit the Neve class.

**Comment by arinaturcu** [comment]
@HardeepAsrani I make the changes, let me know if I missed anything :rocket:

**Comment by HardeepAsrani** [comment]
@arinaturcu Just two changes:

- It might make more sense for the object attributes to be empty so we don't have an align-none class there, it should just be empty when nothing selected.
- If you've one alignment selected (say center) then clicking on it again should bring it back to the default state so user can unselect any selected alignment.

Apart from that, it all seems good and once you make the changes, you can put it for QA.

**Comment by arinaturcu** [comment]
@HardeepAsrani
> * If you've one alignment selected (say center) then clicking on it again should bring it back to the default state so user can unselect any selected alignment.

This already happens, and by unselecting that one alignment, it selects the `none` one.
Also, the `none` alignment doesn't add an `align-none` class, it lets the block without any alignment class and can't be removed because of [this](https://github.com/WordPress/gutenberg/blob/c489f755eed1c6db6bad021562105caa19aad805/packages/block-editor/src/components/block-alignment-control/use-available-alignments.js#L17):
```
	// Always add the `none` option if not exists.
	if ( ! controls.includes( 'none' ) ) {
		controls = [ 'none', ...controls ];
	}
```
Let me know if I am mistaken :see_no_evil:

**Comment by HardeepAsrani** [comment]
@arinaturcu thanks for correcting me. But I would still not like to see a None button as we don't need that, as you'd see in any core blocks.

Also, better to not have any defaults in attributes where possible.

**Comment by arinaturcu** [comment]
@HardeepAsrani The `BlockAlignmentToolbar` doesn't allow to remove the None button, so to remove it I would have the following options:
- use the `AlignmentControl`, but it doesn't have a `full width` option, so I would add a separate control to enable that
- to write it from scratch, using `ToolbarGroup` similar to what I previously did

Let me know which way to proceed, or if you have another suggestion :rocket:

**Comment by HardeepAsrani** [comment]
@arinaturcu I'll see if that is possible but for now I can't really use the PR, it throws an error in console and the block fails.

**Comment by arinaturcu** [comment]
@HardeepAsrani Yes, sorry, forgot to push some changes.

**Comment by irinelenache** [comment]
@arinaturcu Tested and found this:

- [ ] When i try to change the alignment, nothing happens and i'm getting errors in customizer https://vertis.d.pr/rv2WgE

**Comment by arinaturcu** [comment]
Thank you @irinelenache :rocket: 
The issue should be fixed now.

**Comment by irinelenache** [comment]
@arinaturcu Thank you, the error is fixed now 🚀 

Some other findings:

- [ ] The button spacing is not applied in editor when the preview is in tablet/mobile mode https://vertis.d.pr/44snk3
- [ ] The alignments don't work as expected when the buttons are collapsed on mobile, i made a video to better outline the issue https://vertis.d.pr/CqWbpC

You can check the issues here:
```
 Admin area URL: https://otter2.s4-tastewp.com/wp-admin 
 Username: irinel 
 Password: 3hm5qSjv4lc 
```

**Comment by HardeepAsrani** [comment]
> The button spacing is not applied in editor when the preview is in tablet/mobile mode https://vertis.d.pr/44snk3

This is being fixed in the FSE PR already, not related to this PR.

**Comment by arinaturcu** [comment]
@irinelenache The first one is related to another PR as Hardeep said, and the second one should be fixed now.
Thank you! :rocket:

**Comment by irinelenache** [comment]
@arinaturcu Tested again and now everything works fine, will move this to **Ready to merge** 🚀

**Pull request merged**